allennlp/models/archival.py

```diff
@@ -49,6 +49,16 @@ def archive_model(serialization_dir: str,
         archive.add(os.path.join(serialization_dir, "vocabulary"),
                     arcname="vocabulary")
 
+def _sanitize_config(config: Params) -> None:
+    """
+    There are some elements of the model config that we need to get rid of
+    when we load from archive, as they refer to paths on the training machine
+    that are unlikely to exist on the un-archiving machine. To be extra-safe
+    we just remove them from the config.
+    """
+    # TODO(joelgrus): come up with a better solution for this
+    config.get("model", {}).pop('evaluation_json_file', None)
```

Comment:
Yeah, can you check if this exists before removing it?  I definitely want this to be there when running `evaluate` - that's how I knew that my model archive worked.

Other than that you might say in the docstring that this is a temporary fix until we have a better solution for https://github.com/allenai/allennlp/issues/244.

Reply:
Maybe also log a warning when you remove it.